Code/src/main/java/nl/utwente/group10/haskell/type/FuncT.java

```diff
@@ -30,4 +30,10 @@ public final String toHaskellType() {
         out.append(")");
         return out.toString();
     }
+
+    @Override
+    public FuncT getFresh() {
+        Type[] args = getFreshArgs();
+        return new FuncT(args[0], args[1]);
+    }
```

Comment:
It would be very un-Java-like. Furthermore, the responsibility of toHaskell and toString would be in the wrong place - I would be doing ```instanceof``` in ```ConstT``` for its subclasses. That is not nice and does certainly not improve extendibility. So I'm not a big fan of it only for the sake of simplifying its subclasses. If the subclasses are merely a convenience, a factory method would be better anyways.